src/plots/cartesian/tick_label_defaults.js

```diff
@@ -50,6 +50,8 @@ module.exports = function handleTickLabelDefaults(containerIn, containerOut, coe
     }
 
     if(axType !== 'category' && !options.noHover) coerce('hoverformat');
+
+    coerce('ticklabelsautomargin');
```

Comment:
@nicolaskruchten you'll have to add an option to opt-out of coercing `ticklabelautomargin` in `handleTickLabelDefaults` so that other callees 

![image](https://user-images.githubusercontent.com/6675409/35534432-81749e64-050e-11e8-937f-e97decc75076.png)


don't cough on it.